include/arch/arm/aarch32/arch.h

```diff
@@ -40,8 +40,13 @@
 #elif defined(CONFIG_CPU_CORTEX_R) || defined(CONFIG_CPU_AARCH32_CORTEX_A)
 #include <arch/arm/aarch32/cortex_a_r/cpu.h>
 #include <arch/arm/aarch32/cortex_a_r/sys_io.h>
+#if defined(CONFIG_ARMV8_R)
+#include <arch/arm/aarch32/lib_helpers.h>
+#include <arch/arm64/timer.h>
```

Comment:
@carlocaione The arm64 timer can be used here for ARMv8-R AArch32 profile,
does we should move it to a common directory ?

Reply:
uhm, don't think so. The `timer.h` is shipping the arch-level implementation of the `arm_arch_timer_*` functions and those are strictly arch-dependent.

Reply:
ok, I see now that you are using the same timer low level functions. So yeah, either you duplicate the `timer.h` or we can think of moving this to a common directory.

Reply:
Hi, I think we can copy the timer.h in the aarch32 directory. 

About the GIC interface it will be more painful to ifdef all access to:
- ICC_IAR1_EL1 
- ICC_EOIR1_EL1
- ICC_SRE_EL1
- ICC_PMR_EL1
- ICC_IGRPEN1_EL1
In the intc_gicv3.c driver.

Can we keep these 'EL1' macros in lib_helpers  with a comment which explains the
compatibility issue ?

Reply:
Just pushed a new version, aarch32 armv8 now use it's own timer.h, gic v3 driver still use _EL1 macros but I added a comment in lib_helpers.h about them.